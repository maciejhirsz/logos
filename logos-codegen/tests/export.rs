#![cfg(feature = "debug")]

use std::path::Path;

use insta::assert_snapshot;

#[rstest::rstest]
#[case("complex")]
fn test_export(#[case] case: &str) {
    let mut input_path = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/export");
    input_path.push(case);
    input_path.push("input.rs");

    let input = std::fs::read_to_string(input_path).expect("Unable to read input.rs");

    let _ = logos_codegen::generate(input.parse().expect("Unable to parse input.rs"));

    let generated_dot = std::fs::read_to_string(format!("{case}_export_tmp/{case}.dot"))
        .expect("Unable to read dot file");
    let generated_mermaid = std::fs::read_to_string(format!("{case}_export_tmp/{case}.mmd"))
        .expect("Unable to read mermaid file");

    assert_snapshot!(format!("{case}-dot"), generated_dot);
    assert_snapshot!(format!("{case}-mmd"), generated_mermaid);

    // cleanup
    let _ = std::fs::remove_dir_all(format!("{case}_export_tmp"));
}
