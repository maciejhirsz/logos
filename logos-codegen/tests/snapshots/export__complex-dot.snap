---
source: logos-codegen/tests/export.rs
expression: generated_dot
---
digraph {
node[shape=box];
splines=ortho;
n0[label="State 0",color=black];
n0->n1 [label="a..=r|t..=z"];
n0->n2 [label="s"];
n1[label="State 1",color=green];
n2[label="State 2",color=green];
n2->n4 [label="t"];
n3[label="State 3",color=black];
n4[label="State 4",color=black];
n4->n5 [label="r"];
n5[label="State 5",color=black];
n5->n6 [label="\\x00..=`|{..=\\xff"];
n5->n7 [label="a..=t|v..=z"];
n5->n8 [label="u"];
n5->n6 [label="EOI"];
n6[label="State 6",color=green];
n7[label="State 7",color=green];
n7->n9 [label="\\x00..=`|{..=\\xff"];
n7->n10 [label="a..=z"];
n7->n9 [label="EOI"];
n8[label="State 8",color=green];
n8->n9 [label="\\x00..=`|{..=\\xff"];
n8->n10 [label="a..=b|d..=z"];
n8->n11 [label="c"];
n8->n9 [label="EOI"];
n9[label="State 9",color=green];
n10[label="State 10",color=green];
n10->n9 [label="\\x00..=`|{..=\\xff"];
n10->n10 [label="a..=z"];
n10->n9 [label="EOI"];
n11[label="State 11",color=black];
n11->n9 [label="\\x00..=`|{..=\\xff"];
n11->n10 [label="a..=s|u..=z"];
n11->n12 [label="t"];
n11->n9 [label="EOI"];
n12[label="State 12",color=green];
n12->n14 [label="a..=z"];
n13[label="State 13",color=black];
n14[label="State 14",color=black];
n14->n9 [label="\\x00..=`|{..=\\xff"];
n14->n10 [label="a..=z"];
n14->n9 [label="EOI"];
}
